crates/catalog/rest/src/catalog.rs

```diff
@@ -446,6 +490,24 @@ impl RestCatalog {
 
         Ok(())
     }
+
+    fn load_file_io(
+        &self,
+        metadata_location: Option<&str>,
+        extra_config: Option<HashMap<String, String>>,
+    ) -> Result<FileIO> {
+        let mut props = self.config.props.clone();
+        if let Some(config) = extra_config {
+            props.extend(config);
+        }
+
+        let file_io = match self.config.warehouse.as_deref().or(metadata_location) {
+            Some(url) => FileIO::from_path(url)?.with_props(props).build()?,
+            None => FileIOBuilder::new("s3").with_props(props).build()?,
```

Comment:
Do we want to fallback to S3 by default? I would not expect to fallback to S3, and this might lead to unexpected error messages.

Reply:
> I would not expect to fallback to S3, and this might lead to unexpected error messages.

I agree. We shouldn't favor specific storage services.